App/Sagas/TextileSagas.js

```diff
@@ -130,13 +130,18 @@ export function * photosTask (action) {
       yield call(RNFS.unlink, photo.thumbPath)
       photo['hash'] = multipartData.boundary
       yield put(TextileActions.imageAdded(photo, multipartData.payloadPath))
-      yield call(
-        UploadTask.uploadFile,
-        multipartData.payloadPath,
-        'https://ipfs.textile.io/api/v0/add?wrap-with-directory=true',
-        'POST',
-        multipartData.boundary
+      const uploadId = yield call(
+        Upload.startUpload,
+        {
+          path: multipartData.payloadPath,
+          url: 'https://ipfs.textile.io/api/v0/add?wrap-with-directory=true',
+          method: 'POST',
+          type: 'multipart',
+          field: multipartData.boundary
+        }
       )
+      photo['uploadId'] = uploadId
+      yield put(TextileActions.imageSubmittedForUpload(uploadId))
```

Comment:
I think this action should take two args, the photo and the upload id. The reducer that handles it will be responsible for updating the image state to include the id. See my comments in the reducer code.